docs/the-new-architecture/pure-cxx-modules.md

```diff
@@ -0,0 +1,472 @@
+
+1. In Xcode, open the `AppDelegate.mm` file
+2. Modify it as it follows:
+
+```diff
+#import <React/RCTBundleURLProvider.h>
++ #import <RCTAppDelegate+Protected.h>
++ #import "NativeSampleModule.h"
+
+// ...
+  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+#endif
+}
+
++- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const std::string &)name
++                                                      jsInvoker:(std::shared_ptr<facebook::react::CallInvoker>)jsInvoker
++{
++  if (name == "NativeSampleModule") {
++    return std::make_shared<facebook::react::NativeSampleModule>(jsInvoker);
++  }
++
++  return [super getTurboModule:name jsInvoker:jsInvoker];
++}
+
+@end
+```
+
+These changes are doing ta few things:
+
+1. Importing the `RCTAppDelegate+Protected` header to make visible to the AppDelegate that it is conforming to the `RCTTurboModuleManagerDelegate` protocol.
+2. Importing the Pure C++ Native Turbo Module interface `NativeSampleModule.h`
+3. Overriding the `getTurboModule` method for C++ modules so that when the JS side asks for a module called `NativeSampleModule`, the app knows which module has to be returned.
+
+If you now build your application from Xcode, you should be able to build successfully.
+
+## 5. Testing your Code
+
+It's now time to access our C++ Turbo Native Module from JS. To do so, we have to modify the `App.tsx` file to import the Turbo Native Module and to call it in our code.
+
+1. Open the `App.tsx` file
+2. Replace the content of the template with the following code
+
+```ts
+import React from 'react';
+import {
+  Button,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+import SampleTurboModule from './specs/NativeSampleModule';
+
+function App(): React.JSX.Element {
+  const [value, setValue] = React.useState('');
+  const [reversedValue, setReversedValue] = React.useState('');
+
+  const onPress = () => {
+    const revString = SampleTurboModule.reverseString(value);
+    setReversedValue(revString);
+  };
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <View>
+        <Text style={styles.title}>
+          Welcome to C++ Turbo Native Module Example
+        </Text>
+        <Text>Write down here he text you want to revert</Text>
+        <TextInput
+          style={styles.textInput}
+          placeholder="Write your text here"
+          onChangeText={setValue}
+          value={value}
+        />
+        <Button title="Reverse" onPress={onPress} />
+        <Text>Reversed text: {reversedValue}</Text>
+      </View>
+    </SafeAreaView>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  title: {
+    fontSize: 18,
+    marginBottom: 20,
+  },
+  textInput: {
+    borderColor: 'black',
+    borderWidth: 1,
+    borderRadius: 5,
+    padding: 10,
+    marginTop: 10,
+  },
+});
+
```

Comment:
nit: can we simplify this a bit?

Reply:
We could have an hardcoded string, but I feel like it's nice to provide users with something they can play with. :D